src/pages/Dashboard/Modals/BorrowRepay/Borrow/index.tsx

```diff
@@ -0,0 +1,223 @@
+  safeLimitTokens: string;
+  borrow: (amountWei: BigNumber) => Promise<string>;
+  isBorrowLoading: boolean;
+}
+
+export const BorrowForm: React.FC<IBorrowFormProps> = ({
+  asset,
+  limitTokens,
+  safeBorrowLimitPercentage,
+  safeLimitTokens,
+  borrow,
+  isBorrowLoading,
+}) => {
+  const { t } = useTranslation();
+  const sharedStyles = useStyles();
+  const borrowStyles = useBorrowStyles();
+  const styles = {
+    ...sharedStyles,
+    ...borrowStyles,
+  };
+
+  const { openSuccessfulTransactionModal } = useSuccessfulTransactionModal();
+
+  const onSubmit: IAmountFormProps['onSubmit'] = async amountTokens => {
+    const formattedAmountTokens = new BigNumber(amountTokens);
+
+    const amountWei = convertCoinsToWei({
+      value: formattedAmountTokens,
+      tokenId: asset.id,
+    });
+
+    try {
+      // Send request to repay VAI
+      const transactionHash = await borrow(amountWei);
+
+      // Display successful transaction modal
+      openSuccessfulTransactionModal({
+        title: t('borrowRepayModal.borrow.successfulTransactionModal.title'),
+        message: t('borrowRepayModal.borrow.successfulTransactionModal.message'),
+        amount: {
+          valueWei: amountWei,
+          tokenId: asset.id,
+        },
+        transactionHash,
+      });
+    } catch (error) {
+      toast.error({ title: (error as Error).message });
+    }
+  };
+
+  return (
+    <AmountForm onSubmit={onSubmit} maxAmount={limitTokens}>
+      {({ values, setFieldValue, handleBlur, dirty, isValid, errors }) => (
+        <>
+          <div css={[styles.getRow({ isLast: true })]}>
+            <TokenTextField
+              name="amount"
+              tokenId={asset.id}
+              value={values.amount}
+              onChange={amount => setFieldValue('amount', amount, true)}
+              disabled={isBorrowLoading}
+              onBlur={handleBlur}
+              rightMaxButton={{
+                label: t('borrowRepayModal.borrow.rightMaxButtonLabel', {
+                  limitPercentage: safeBorrowLimitPercentage,
+                }),
+                valueOnClick: safeLimitTokens,
+              }}
+              data-testid="token-text-field"
+              // Only display error state if amount is higher than borrow limit
+              hasError={errors.amount === ErrorCode.HIGHER_THAN_MAX}
+            />
+
+            {+values.amount > +safeLimitTokens && (
+              <div css={styles.liquidationWarning}>
+                <Icon name="info" css={styles.liquidationWarningIcon} />
+
+                <Typography variant="small2" css={styles.whiteLabel}>
+                  {t('borrowRepayModal.borrow.highAmountWarning')}
+                </Typography>
+              </div>
+            )}
+          </div>
+
+          <AccountData amount={values.amount} asset={asset} />
+
+          <PrimaryButton
+            type="submit"
+            loading={isBorrowLoading}
+            disabled={!isValid || !dirty || isBorrowLoading}
```

Comment:
Id think the case where the firm is valid it would always be dirty and when it is not dirty it is always invalid. Is there a specific case where we need to check for the form being valid  but not dirty?

Reply:
Logically speaking, you're right. But the issue is that the validation is asynchronous, so when it happens on mount the form is first marked as `valid` until the validation has happened.
Adding `dirty` means the button is disabled even before the first validation has occurred.

Reply:
Interesting our validation shouldn't be async and we have `validateOnMount` set so this behaviour isn't what I would expect. I checked it out and I see what you mean though

Reply:
I agree with you. I tried a few things, every time it seemed the validation on mount would take some time during which the button wouldn't be disabled :/